diff --git a/lib/index.d.ts b/lib/index.d.ts
index 83f86fa18f74bbedc4800f392a4d99ce53bb5fce..f3ffa5eaa2de45a5cfa397dc4bd9ce8888f3f978 100644
--- a/lib/index.d.ts
+++ b/lib/index.d.ts
@@ -23,7 +23,9 @@ interface SourceLocation {
         column: number;
     };
 }
-interface BaseNode {
+
+// We export `BaseNode` so that we can augment it on our end.
+export interface BaseNode {
     type: Node["type"];
     leadingComments?: Comment[] | null;
     innerComments?: Comment[] | null;
