diff --git a/node_modules/eslint/lib/linter/node-event-generator.js b/node_modules/eslint/lib/linter/node-event-generator.js
index 6f3b251..531e463 100644
--- a/node_modules/eslint/lib/linter/node-event-generator.js
+++ b/node_modules/eslint/lib/linter/node-event-generator.js
@@ -9,8 +9,14 @@
 // Requirements
 //------------------------------------------------------------------------------
 
-const esquery = require("esquery");
-const lodash = require("lodash");
+let esquery = require('esquery');
+const lodash = require('lodash');
+
+// When building eslint for browser, esquery gets wrapped in an esmodule
+// under the `default` key = so we just need to check and swap
+if (esquery.default) {
+  esquery = esquery.default;
+}
 
 //------------------------------------------------------------------------------
 // Typedefs
